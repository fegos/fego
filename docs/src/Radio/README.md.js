webpackJsonp([7],{471:function(a,n){a.exports={content:["article",["h1","\u5355\u9009\u6846 Radio"],["h3","\u7ec4\u4ef6\u63cf\u8ff0"],["ul",["li",["p","\u540c\u4e8e\u5411\u7528\u6237\u63d0\u4f9b\u5355\u9009\u9009\u62e9\u3002"]]],["h3","\u793a\u4f8b\u4ee3\u7801"],["pre",{lang:"html",highlighted:'<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">defaultChecked</span><span class="token punctuation">></span></span>\u9ed8\u8ba4\u9009\u4e2d\u7684\u5355\u9009\u6846<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio.Group</span> <span class="token attr-name">name</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>rbox<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>chi<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u8bed\u6587<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>mat<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u6570\u5b66<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>eng<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u82f1\u8bed<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>phy<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u7269\u7406<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>che<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u5316\u5b66<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>Radio</span> <span class="token attr-name">value</span><span class="token attr-value"><span class="token punctuation">=</span><span class="token punctuation">"</span>bio<span class="token punctuation">"</span></span><span class="token punctuation">></span></span>\u751f\u7269<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio</span><span class="token punctuation">></span></span>\n<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>Radio.Group</span><span class="token punctuation">></span></span>'},["code",'<Radio defaultChecked>\u9ed8\u8ba4\u9009\u4e2d\u7684\u5355\u9009\u6846</Radio>\n\n<Radio.Group name="rbox">\n  <Radio value="chi">\u8bed\u6587</Radio>\n  <Radio value="mat">\u6570\u5b66</Radio>\n  <Radio value="eng">\u82f1\u8bed</Radio>\n  <Radio value="phy">\u7269\u7406</Radio>\n  <Radio value="che">\u5316\u5b66</Radio>\n  <Radio value="bio">\u751f\u7269</Radio>\n</Radio.Group>']],["h2","API"],["h3","Radio props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","label"],["td","\u5355\u9009\u6846\u6587\u6848"],["td","string"],["td"]],["tr",["td","children"],["td","\u5355\u9009\u6846\u6587\u6848\uff0c\u4f7f\u7528\u65f6label\u5c5e\u6027\u5931\u6548"],["td","react node"],["td"]],["tr",["td","checked"],["td","\u662f\u5426\u9009\u4e2d\uff0c\u53d7\u63a7\u5c5e\u6027"],["td","boolean"],["td"]],["tr",["td","defaultChecked"],["td","\u662f\u5426\u9009\u4e2d\uff0c\u975e\u53d7\u63a7\u5c5e\u6027"],["td","boolean"],["td"]],["tr",["td","disabled"],["td","\u662f\u5426\u7981\u7528"],["td","boolean"],["td","false"]],["tr",["td","onChange"],["td","\u53d8\u5316\u56de\u8c03"],["td","function(e){}"],["td"]]]],["h3","Radio.Group props\u5217\u8868"],["table",["thead",["tr",["th","\u5c5e\u6027"],["th","\u8bf4\u660e"],["th","\u7c7b\u578b"],["th","\u9ed8\u8ba4\u503c"]]],["tbody",["tr",["td","name"],["td","Group \u4e0b\u6240\u6709 input",["span",'type="Radio"']," \u7684 name \u5c5e\u6027"],["td","string"],["td"]],["tr",["td","value"],["td","\u5f53\u524d\u9009\u4e2d\u7684\u5355\u9009\u6846\u7684\u503c\uff0c\u53d7\u63a7\u5c5e\u6027"],["td","string"],["td"]],["tr",["td","defaultValue"],["td","\u5f53\u524d\u9009\u4e2d\u7684\u5355\u9009\u6846\u7684\u503c\uff0c\u975e\u53d7\u63a7\u5c5e\u6027"],["td","string"],["td"]],["tr",["td","onChange"],["td","\u53d8\u5316\u56de\u8c03"],["td","function(e){}"],["td"]],["tr",["td","options"],["td","\u4ee5\u914d\u7f6e\u5f62\u5f0f\u8bbe\u7f6e\u5b50\u5143\u7d20"],["td","object array"],["td"]]]]],meta:{title:"Radio",subTitle:"\u5355\u9009\u6846",filename:"src/Radio/README.md"},toc:["ul",["li",["a",{className:"bisheng-toc-h1",href:"#\u5355\u9009\u6846-Radio",title:"\u5355\u9009\u6846 Radio"},"\u5355\u9009\u6846 Radio"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u7ec4\u4ef6\u63cf\u8ff0",title:"\u7ec4\u4ef6\u63cf\u8ff0"},"\u7ec4\u4ef6\u63cf\u8ff0"]],["li",["a",{className:"bisheng-toc-h3",href:"#\u793a\u4f8b\u4ee3\u7801",title:"\u793a\u4f8b\u4ee3\u7801"},"\u793a\u4f8b\u4ee3\u7801"]],["li",["a",{className:"bisheng-toc-h2",href:"#API",title:"API"},"API"]],["li",["a",{className:"bisheng-toc-h3",href:"#Radio-props\u5217\u8868",title:"Radio props\u5217\u8868"},"Radio props\u5217\u8868"]],["li",["a",{className:"bisheng-toc-h3",href:"#Radio.Group-props\u5217\u8868",title:"Radio.Group props\u5217\u8868"},"Radio.Group props\u5217\u8868"]]]}}});