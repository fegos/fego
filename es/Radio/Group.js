'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var RadioGroup = function (_Component) {
  _inherits(RadioGroup, _Component);

  function RadioGroup(props) {
    _classCallCheck(this, RadioGroup);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.onChange = function (e) {
      var value = e.target.value;
      var onChange = _this.props.onChange;

      if (!('value' in _this.props)) {
        _this.setState({
          value: value
        });
      }
      // onChange instanceof Function && onChange(e)
      onChange instanceof Function && onChange(value);
    };

    _this.state = {
      value: 'value' in props ? props.value : props.defaultValue
    };
    return _this;
  }

  RadioGroup.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('value' in nextProps && nextProps.value !== this.props.value) {
      this.setState({
        value: nextProps.value
      });
    }
  };

  RadioGroup.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        prefixCls = _props.prefixCls,
        className = _props.className,
        style = _props.style,
        name = _props.name,
        children = _props.children,
        options = _props.options;

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('' + prefixCls, className), style: style },
      children ? _react2.default.Children.map(children, function (child) {
        return _react2.default.cloneElement(child, {
          name: name,
          checked: child.props.value === _this2.state.value,
          onChange: _this2.onChange
        });
      }) : options.map(function (opt) {
        return _react2.default.createElement(
          _index2.default,
          {
            key: opt.value,
            value: opt.value,
            name: name,
            checked: opt.value === _this2.state.value,
            onChange: _this2.onChange,
            disabled: opt.disabled
          },
          opt.label
        );
      })
    );
  };

  return RadioGroup;
}(_react.Component);

RadioGroup.defaultProps = {
  // 前缀
  prefixCls: 'ns-radio-group'
};
RadioGroup.propTypes = {
  // 前缀
  prefixCls: _propTypes2.default.oneOf(['ns-radio-group']),
  // RadioGroup 下所有 input[type="radio"] 的 name 属性
  name: _propTypes2.default.string.isRequired,
  // 受控属性，当前选中的值
  value: _propTypes2.default.string,
  // 非受控属性，默认选中的值
  defaultValue: _propTypes2.default.string,
  // 选项变化时的回调函数
  onChange: _propTypes2.default.func,
  // 以配置形式设置子元素
  options: _propTypes2.default.arrayOf(_propTypes2.default.object),
  // 子元素
  children: _propTypes2.default.node
};
exports.default = RadioGroup;