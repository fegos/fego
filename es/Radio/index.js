'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Group = require('./Group');

var _Group2 = _interopRequireDefault(_Group);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Radio = function (_Component) {
  _inherits(Radio, _Component);

  function Radio(props) {
    _classCallCheck(this, Radio);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.onChange = function (e) {
      var _this$props = _this.props,
          onChange = _this$props.onChange,
          disabled = _this$props.disabled;
      var checked = e.nativeEvent.target.checked;


      if (disabled) return;

      if (!('checked' in _this.props)) {
        _this.setState({
          checked: checked
        });
      }

      onChange instanceof Function && onChange(e.nativeEvent);
    };

    _this.state = {
      checked: 'checked' in props ? props.checked : props.defaultChecked
    };
    return _this;
  }

  Radio.prototype.componentWillReceiveProps = function componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps && nextProps.checked !== this.props.checked) {
      this.setState({
        checked: nextProps.checked
      });
    }
  };

  Radio.prototype.renderLabel = function renderLabel() {
    var _props = this.props,
        prefixCls = _props.prefixCls,
        label = _props.label,
        children = _props.children;


    if (children) {
      return _react2.default.createElement(
        'span',
        { className: prefixCls + '-label' },
        children
      );
    } else if (label) {
      return _react2.default.createElement(
        'span',
        { className: prefixCls + '-label' },
        label
      );
    }
    return null;
  };

  Radio.prototype.render = function render() {
    var _classNames,
        _this2 = this;

    var _props2 = this.props,
        prefixCls = _props2.prefixCls,
        className = _props2.className,
        style = _props2.style,
        children = _props2.children,
        disabled = _props2.disabled,
        otherProps = _objectWithoutProperties(_props2, ['prefixCls', 'className', 'style', 'children', 'disabled']);

    return _react2.default.createElement(
      'label',
      {
        htmlFor: otherProps.id,
        className: (0, _classnames2.default)(prefixCls + '-wrapper', className),
        style: style
      },
      _react2.default.createElement(
        'span',
        { className: (0, _classnames2.default)('' + prefixCls, (_classNames = {}, _classNames[prefixCls + '-checked'] = this.state.checked, _classNames[prefixCls + '-disabled'] = disabled, _classNames))
        },
        _react2.default.createElement('input', _extends({
          type: 'radio',
          ref: function ref(i) {
            _this2.radioRef = i;
          }
        }, otherProps, {
          disabled: disabled,
          className: prefixCls + '-ipt',
          onChange: this.onChange
        })),
        _react2.default.createElement(
          'span',
          { className: prefixCls + '-inner' },
          _react2.default.createElement('span', { className: prefixCls + '-inner-circle' })
        )
      ),
      this.renderLabel()
    );
  };

  return Radio;
}(_react.Component);

Radio.defaultProps = {
  // 前缀
  prefixCls: 'ns-radio'
};
Radio.propTypes = {
  // 前缀
  prefixCls: _propTypes2.default.oneOf(['ns-radio']),
  // radio label 文案
  label: _propTypes2.default.string,
  // radio label 文案，使用则 labe 属性失效
  children: _propTypes2.default.node,
  // 受控属性，当前是否选中，需配合 onChane 使用
  checked: _propTypes2.default.bool,
  // 回调
  onChange: _propTypes2.default.func,
  // 非受控属性，当前是否选中
  defaultChecked: _propTypes2.default.bool,
  // 是否禁用
  disabled: _propTypes2.default.bool
};
exports.default = Radio;


Radio.Group = _Group2.default;