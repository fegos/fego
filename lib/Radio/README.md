---
title: Radio
subTitle: 单选框
---

# 单选框 Radio

### 组件描述
- 同于向用户提供单选选择。

### 示例代码

```html
<Radio defaultChecked>默认选中的单选框</Radio>

<Radio.Group name="rbox">
  <Radio value="chi">语文</Radio>
  <Radio value="mat">数学</Radio>
  <Radio value="eng">英语</Radio>
  <Radio value="phy">物理</Radio>
  <Radio value="che">化学</Radio>
  <Radio value="bio">生物</Radio>
</Radio.Group>
```

## API

### Radio props列表

| 属性 | 说明 | 类型 | 默认值 |
|----|-----|------|------|
| label | 单选框文案 | string |  |
| children | 单选框文案，使用时label属性失效 | react node |  |
| checked | 是否选中，受控属性 | boolean |  |
| defaultChecked | 是否选中，非受控属性 | boolean |  |
| disabled | 是否禁用 | boolean | false |
| onChange | 变化回调 | function(e){} |  |


### Radio.Group props列表

| 属性 | 说明 | 类型 | 默认值 |
|----|-----|------|------|
| name | Group 下所有 input[type="Radio"] 的 name 属性 | string |  |
| value | 当前选中的单选框的值，受控属性 | string |  |
| defaultValue | 当前选中的单选框的值，非受控属性 | string |  |
| onChange | 变化回调 | function(e){} |  |
| options | 以配置形式设置子元素 | object array |  |
