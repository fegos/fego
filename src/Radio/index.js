import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import RadioGroup from './Group';

export default class Radio extends Component {
  static defaultProps = {
    // 前缀
    prefixCls: 'ns-radio',
  }

  static propTypes = {
    // 前缀
    prefixCls: PropTypes.oneOf(['ns-radio']),
    // radio label 文案
    label: PropTypes.string,
    // radio label 文案，使用则 labe 属性失效
    children: PropTypes.node,
    // 受控属性，当前是否选中，需配合 onChane 使用
    checked: PropTypes.bool,
    // 回调
    onChange: PropTypes.func,
    // 非受控属性，当前是否选中
    defaultChecked: PropTypes.bool,
    // 是否禁用
    disabled: PropTypes.bool,
  }

  constructor(props) {
    super(props);

    this.state = {
      checked: 'checked' in props ? props.checked : props.defaultChecked,
    };
  }

  componentWillReceiveProps(nextProps) {
    if ('checked' in nextProps && nextProps.checked !== this.props.checked) {
      this.setState({
        checked: nextProps.checked,
      });
    }
  }

  onChange = (e) => {
    const { onChange, disabled } = this.props;
    const { checked } = e.nativeEvent.target;

    if (disabled) return;

    if (!('checked' in this.props)) {
      this.setState({
        checked,
      });
    }

    onChange instanceof Function && onChange(e.nativeEvent);
  }

  renderLabel() {
    const { prefixCls, label, children } = this.props;

    if (children) {
      return <span className={`${prefixCls}-label`}>{children}</span>;
    } else if (label) {
      return <span className={`${prefixCls}-label`}>{label}</span>;
    }
    return null;
  }

  render() {
    const {
      prefixCls, className, style, children, disabled, ...otherProps
    } = this.props;

    return (
      <label
        htmlFor={otherProps.id}
        className={classNames(`${prefixCls}-wrapper`, className)}
        style={style}
      >
        <span className={classNames(`${prefixCls}`, {
          [`${prefixCls}-checked`]: this.state.checked,
          [`${prefixCls}-disabled`]: disabled,
        })}
        >
          <input
            type="radio"
            ref={(i) => { this.radioRef = i; }}
            {...otherProps}
            disabled={disabled}
            className={`${prefixCls}-ipt`}
            onChange={this.onChange}
          />
          <span className={`${prefixCls}-inner`}>
            <span className={`${prefixCls}-inner-circle`} />
          </span>
        </span>
        {this.renderLabel()}
      </label>
    );
  }
}

Radio.Group = RadioGroup;
