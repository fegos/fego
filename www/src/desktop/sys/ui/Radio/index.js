import style from './index.less'
import React, { Component } from 'react'
import { Radio } from 'fego'

export default class Page extends Component {
	constructor(props) {
		super(props)

		this.state = {
			radio1: true,
			radio2: false,
			radioGroupValue1: 'mat',
			radioGroupValue2: 'bio',
		}
	}
	render() {
		let opt = [
			<Radio value='chi' key='0'>语文</Radio>,
			<Radio value='mat' key='1'>数学</Radio>,
			<Radio value='eng' key='2'>英语</Radio>,
			<Radio value='phy' key='3'>物理</Radio>,
			<Radio value='che' key='4'>化学</Radio>,
			<Radio value='bio' key='5'>生物</Radio>
		];

		let opt2 = [
			{ label: '语文', value: 'chi' },
			{ label: '数学', value: 'mat' },
			{ label: '英语', value: 'eng', disabled: true },
			{ label: '物理', value: 'phy'},
			{ label: '化学', value: 'che'},
			{ label: '生物', value: 'bio'},
		];

		return (
			<div className={style.page}>
				<div className='section'>
					<h3>单独使用的 Radio: </h3>
					<Radio defaultChecked={true} onChange={(e)=>{console.log(e.target.checked)}}>使用defaultChecked=true</Radio>
					<Radio value='1' defaultChecked={false} onChange={(e)=>{console.log(e.target.checked,e.target.value)}}>使用defaultChecked=false</Radio>
					<Radio checked={this.state.radio1} onChange={(e)=>{console.log(e.target.checked);this.setState({radio1: e.target.checked})}}>使用checked=true</Radio>
					<Radio value='2' checked={this.state.radio2} onChange={(e)=>{console.log(e.target.checked,e.target.value);this.setState({radio2: e.target.checked})}}>使用checked=false</Radio>
					<Radio disabled checked={true} >disabled</Radio>
					<Radio disabled checked={false} >disabled</Radio>
				</div>

				<div className='section'>
					<h3>Radio.Group: </h3>
					<Radio.Group name='course1' onChange={(value)=>{console.log(value)}}>
						{opt}
					</Radio.Group>
				</div>

				<div className='section'>
					<h3>Radio.Group，使用非受控属性 defaultValue: </h3>
					<Radio.Group name='course2' defaultValue='che' onChange={(value)=>{console.log(value)}}>
						{opt}
					</Radio.Group>
				</div>

				<div className='section'>
					<h3>Radio.Group，使用受控属性 value: </h3>
					<Radio.Group name='course3' value={this.state.radioGroupValue1} onChange={(value)=>{this.setState({
						radioGroupValue1: value
					})}}>
						{opt}
					</Radio.Group>
				</div>

				<div className='section'>
					<h3>Radio.Group，以配置项形式设置子元素: </h3>
					<Radio.Group name='course4' options={opt2} value={this.state.radioGroupValue2} onChange={(value)=>{this.setState({
						radioGroupValue2: value
					})}} />
				</div>
			</div>
		)
	}
}
