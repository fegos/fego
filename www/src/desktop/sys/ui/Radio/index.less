.page {
  padding: 20px 10px;
  :global {
    .section {
      margin-top: 20px;
      h3 {
        margin-bottom: 5px;
      }
    }
  }
}
