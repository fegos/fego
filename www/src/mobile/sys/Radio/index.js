import React, { Component } from 'react';
import { Radio } from 'fego';

export default class Page extends Component {
  state = {
    radio1: true,
    radio2: false,
    radioGroupValue1: 'mat',
    radioGroupValue2: 'bio',
  }
  render() {
    const opt = [
      <Radio value="chi" key="0">语文</Radio>,
      <Radio value="mat" key="1">数学</Radio>,
      <Radio value="eng" key="2">英语</Radio>,
      <Radio value="phy" key="3">物理</Radio>,
      <Radio value="che" key="4">化学</Radio>,
      <Radio value="bio" key="5">生物</Radio>,
    ];
    const opt2 = [
      { label: '语文', value: 'chi' },
      { label: '数学', value: 'mat' },
      { label: '英语', value: 'eng', disabled: true },
      { label: '物理', value: 'phy' },
      { label: '化学', value: 'che' },
      { label: '生物', value: 'bio' },
    ];

    return (
      <div>
        <p className="title">单独使用的 Radio</p>
        <div><Radio defaultChecked onChange={(e) => { console.log(e.target.checked); }}>使用 defaultChecked = true</Radio></div>
        <div><Radio value="1" defaultChecked={false} onChange={(e) => { console.log(e.target.checked, e.target.value); }}>使用 defaultChecked = false</Radio></div>
        <div><Radio checked={this.state.radio1} onChange={(e) => { console.log(e.target.checked); this.setState({ radio1: e.target.checked }); }}>使用 checked = true</Radio></div>
        <div><Radio value="2" checked={this.state.radio2} onChange={(e) => { console.log(e.target.checked, e.target.value); this.setState({ radio2: e.target.checked }); }}>使用 checked = false</Radio></div>
        <div><Radio disabled checked >disabled checked radio</Radio></div>
        <div><Radio disabled checked={false} >disabled unchecked radio</Radio></div>

        <p className="title">组合使用的 Radio，不带 defaultValue: </p>
        <Radio.Group name="course1" onChange={(value) => { console.log(value); }}>
          {opt}
        </Radio.Group>

        <p className="title">组合使用的 Radio，带 defaultValue: </p>
        <Radio.Group name="course2" defaultValue="che" onChange={(value) => { console.log(value); }}>
          {opt}
        </Radio.Group>

        <p className="title">组合使用的 Radio，带 value: </p>
        <Radio.Group
          name="course3"
          value={this.state.radioGroupValue1}
          onChange={(value) => {
            this.setState({
              radioGroupValue1: value,
            });
          }}
        >
          {opt}
        </Radio.Group>

        <p className="title">组合使用的 Radio，以配置项形式设置子元素: </p>
        <Radio.Group
          name="course4"
          options={opt2}
          value={this.state.radioGroupValue2}
          onChange={(value) => {
          this.setState({
            radioGroupValue2: value,
          });
        }}
        />
      </div>
    );
  }
}
