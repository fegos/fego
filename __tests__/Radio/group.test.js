import React from 'react';
import Renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import Radio from 'Radio';

const { Group } = Radio;

describe('Radio Group test', () => {
  let opts = [];
  beforeEach(() => {
    opts = [
      <Radio value="chi" key="0">语文</Radio>,
      <Radio value="mat" key="1">数学</Radio>,
      <Radio value="eng" key="2">英语</Radio>,
      <Radio value="phy" key="3">物理</Radio>,
      <Radio value="che" key="4">化学</Radio>,
      <Radio value="bio" key="5">生物</Radio>,
    ];
  });
  it('radio group snapshot', () => {
    const group = Renderer.create((
      <Group name="course1">
        {opts}
      </Group>
    ));
    expect(group.toJSON()).toMatchSnapshot();
  });

  it('radio group with options should also render correctly', () => {
    const options = [
      { label: '语文', value: 'chi' },
      { label: '数学', value: 'mat' },
      { label: '英语', value: 'eng', disabled: true },
      { label: '物理', value: 'phy' },
      { label: '化学', value: 'che' },
      { label: '生物', value: 'bio' },
    ];
    const group = shallow(<Group name="course2" options={options} />);
    expect(group.find(Radio)).toHaveLength(options.length);
  });

  it('prop value changes should trigger componentWillReceiveProps method', () => {
    const group = shallow((
      <Group name="course3" value="chi">
        {opts}
      </Group>
    ));
    const spy = sinon.spy(Group.prototype, 'componentWillReceiveProps');

    group.setProps({
      value: 'phy',
    });

    expect(spy.callCount).toBe(1);
  });

  it('radio onChange event should trigger radioGroup with value prop onChange callback', () => {
    const oldGroupValue = '';
    const newRadioValue = true;
    const spyRadioOnChange = sinon.spy((e) => {
      expect(spyRadioOnChange.callCount).toEqual(1);
      expect(e.target.checked).toEqual(newRadioValue);
    });
    const spyRadioGroupOnChange = sinon.spy(() => {
      expect(spyRadioGroupOnChange.callCount).toEqual(1);
    });

    const group = mount((
      <Group name="course4" value={oldGroupValue} onChange={spyRadioGroupOnChange}>
        <Radio onChange={spyRadioOnChange} value="0">radio</Radio>
      </Group>
    ));

    group.find('input').simulate('change', {
      nativeEvent: {
        target: {
          checked: newRadioValue,
        },
      },
    });
  });

  it('radio onChange event should trigger radioGroup with defaultValue prop onChange callback', () => {
    const oldGroupValue = '';
    const newRadioValue = false;
    const spyRadioOnChange = sinon.spy((e) => {
      expect(spyRadioOnChange.callCount).toEqual(1);
      expect(e.target.checked).toEqual(newRadioValue);
    });
    const spyRadioGroupOnChange = sinon.spy(() => {
      expect(spyRadioGroupOnChange.callCount).toEqual(1);
    });

    const group = mount((
      <Group name="course5" defaultValue={oldGroupValue} onChange={spyRadioGroupOnChange}>
        <Radio onChange={spyRadioOnChange} value="0">radio</Radio>
      </Group>
    ));

    group.find('input').simulate('change', {
      nativeEvent: {
        target: {
          checked: newRadioValue,
        },
      },
    });
  });
});
