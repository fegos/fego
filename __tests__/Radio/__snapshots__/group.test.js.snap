// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Group test radio group snapshot 1`] = `
<div
  className="ns-radio-group"
  style={undefined}
>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="chi"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      语文
    </span>
  </label>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="mat"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      数学
    </span>
  </label>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="eng"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      英语
    </span>
  </label>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="phy"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      物理
    </span>
  </label>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="che"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      化学
    </span>
  </label>
  <label
    className="ns-radio-wrapper"
    htmlFor={undefined}
    style={undefined}
  >
    <span
      className="ns-radio"
    >
      <input
        checked={false}
        className="ns-radio-ipt"
        disabled={undefined}
        name="course1"
        onChange={[Function]}
        type="radio"
        value="bio"
      />
      <span
        className="ns-radio-inner"
      >
        <span
          className="ns-radio-inner-circle"
        />
      </span>
    </span>
    <span
      className="ns-radio-label"
    >
      生物
    </span>
  </label>
</div>
`;
